Add concepts: system package managers (#242)
microsoft/vcpkg-docs: vcpkg/concepts/system-package-managers.md

---

> title: System package managers

Reviewer: ```suggestion
title: Using vcpkg compared to system package managers
```

description: Concepts on system package managers
author: data-queue
ms.author: danshaw2
ms.date: 01/10/2024
ms.prod: vcpkg
ms.topic: concept-article
---

# System package managers

> C++ libraries are available in system package managers, such as apt, and vcpkg. How you choose to acquire your libraries depends on your specific usecase.

Reviewer: ```suggestion
C++ libraries are available in system package managers such as apt and vcpkg. How you choose to acquire your libraries depends on your specific use case.
```

Reviewer 2: ```suggestion
C++ libraries are available in system package managers, such as apt, and in vcpkg. How you choose to acquire your libraries depends on your specific use case.
```
I initially read "such as apt and vcpkg", but this isn't what is meant...

> ## Pros of using dependencies from system package manager

Reviewer: ```suggestion
## Reasons to use dependencies from a system package manager
```

> ## Pros of using dependencies from system package manager

Reviewer 3: It would be good to link to https://devblogs.microsoft.com/cppblog/using-system-package-manager-dependencies-with-vcpkg/ from here (or copy that blog post effectively to `learn.`) so that readers have any idea what it means to do this.

> - Overall deployment size will be reduced for customers as different programs can share the same share libraries.

Reviewer: ```suggestion
- Overall deployment size will be reduced, as different programs can share the same shared libraries.
```

- Your program will respond to policy decisions configured on that system. For example, DNS policy tends to be implemented in glibc, or trusted certificates implemented in OpenSSL, or UI / themeing bits from UI libraries, etc.
- It's usually faster to download binaries from a system package manager than it is to build them from source.

> ## Pros of using dependencies through vcpkg

Reviewer: ```suggestion
## Reasons to use dependencies from vcpkg
```

- It will easier for development: vcpkg builds from source, meaning you will have debug information for libraries. Additionally, libraries build both release and debug versions.

> - You have any control whatsoever over the versions of those dependencies. With a system package manager you're usually stuck with whatever versions it gives you which often forces folks to make different distributions of their code for each distro and version of distro. (In particular, this means you can get newer dependencies)

Reviewer: ```suggestion
- You have complete control over the versions of those dependencies. With a system package manager, you may be limited to only certain versions.
```

> - The way you get dependencies might be the same for all platforms instead of needing carve outs for different systems.

Reviewer: ```suggestion
- The way you get dependencies can be the same for all platforms that you target.
```
